mirage/mimic_happy_eyeballs.mli

```diff
@@ -30,7 +30,8 @@ end
     user. *)
 module Make
     (Stack : Tcpip.Stack.V4V6)
-    (Happy_eyeballs : Happy_eyeballs_mirage.S with type flow = Stack.TCP.flow) : sig
+    (Happy_eyeballs : Happy_eyeballs_mirage.S with type flow = Stack.TCP.flow)
+    (_ : Dns_client_mirage.S) : sig
```

Comment:
I'm not entirely sure, do we want some type equality such as `with type happy_eyeballs = Happy_eyeballs.t and type Transport.stack = Stack.t * Happy_eyeballs.t`?

Reply:
I added these in #25.